src/sfizz/FilePool.cpp

```diff
@@ -46,27 +46,114 @@
 #endif
 using namespace std::placeholders;
 
-static std::weak_ptr<ThreadPool> globalThreadPoolWeakPtr;
-static std::mutex globalThreadPoolMutex;
-
-static std::shared_ptr<ThreadPool> globalThreadPool()
+sfz::FilePool::GlobalObject::GlobalObject(size_t num_threads)
+: lastGarbageCollection_ { highResNow() },
+  threadPool { new ThreadPool(num_threads) }
 {
-    std::shared_ptr<ThreadPool> threadPool;
+    garbageThread.reset(new std::thread(&GlobalObject::garbageJob, this));
+}
 
-    threadPool = globalThreadPoolWeakPtr.lock();
-    if (threadPool)
-        return threadPool;
+sfz::FilePool::GlobalObject::~GlobalObject()
+{
+    garbageFlag = false;
+    std::error_code ec;
+    semGarbageBarrier.post(ec);
+    ASSERT(!ec);
+    garbageThread->join();
+    // clear semaphore
+    while (semGarbageBarrier.try_wait());
+}
 
-    std::lock_guard<std::mutex> lock(globalThreadPoolMutex);
-    threadPool = globalThreadPoolWeakPtr.lock();
-    if (threadPool)
-        return threadPool;
+RTSemaphore sfz::FilePool::GlobalObject::semGarbageBarrier { 0 };
+std::weak_ptr<sfz::FilePool::GlobalObject> sfz::FilePool::globalObjectWeakPtr;
+std::mutex sfz::FilePool::globalObjectMutex;
 
-    unsigned numThreads = std::thread::hardware_concurrency();
+std::shared_ptr<sfz::FilePool::GlobalObject> sfz::FilePool::getGlobalObject()
+{
+    std::lock_guard<std::mutex> lock { globalObjectMutex };
+    
+    std::shared_ptr<GlobalObject> globalObject = globalObjectWeakPtr.lock();
+    if (globalObject)
+        return globalObject;
+
+#if 1
+    constexpr size_t numThreads = 1;
```

Comment:
That change can be baked in. iirc sforzando also seems to not try and parallelize reads. There are little situations where this would seem worthwile to do so.

Reply:
Previous:
sfizz loads sample files at parallel threads. once it starts loading, never stops until it finishes.

This PR:
sfizz loads samples files on sliced jobs at a single thread. the files to load is cyclic.

discussion:
parallel disk access is really slow. i received a report that shows me that terrible noises occurs on M1 iPad Pro when the person uses isfizz with other samplers. I guessed it is from disk access and it gets better.

Reply:
Yes that's a much better approach. I also think we could do this separately from the rest of the PR.